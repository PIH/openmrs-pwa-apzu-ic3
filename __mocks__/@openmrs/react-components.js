import types from 'redux-types';
// mock methods

const visitRest = {

  createVisit: (params) => {

    const visit = params.visit;

    if (visit.patient != 'patient_uuid' || visit.visitType != 'visit_type_uuid' || visit.location != 'location_uuid') {
      throw {
        name: 'Mock Exception',
        message: 'Unable to Submit'
      };
    }
    else {
      return;
    }
  }

};

const encounterRest = {

  createEncounter: (params) => {

    if (params.encounter.encounterType === 'invalid_encounter_type') {
      throw {
        name: 'Mock Exception',
        message: 'Unable to Submit'
      };
    }
    else {
      return;
    }
  }

};

const BASIC_TYPES = [
  'FETCH_SUCCEEDED',
  'FETCH_PENDING',
  'SET_SUCCEEDED',
  'SET_REQUESTED',
  'SET_FAILED',
  'FETCH_FAILED',
  'FETCH_REQUESTED',
  'SUCCEEDED',
  'PENDING',
  'FAILED',
  'REQUESTED'
];

const LOGIN_TYPES = {
  LOGIN: types('login', BASIC_TYPES),
  LOGIN_LOCATIONS: types('login_locations', BASIC_TYPES),
  LOGOUT: types('logout', BASIC_TYPES)
}

const SESSION_TYPES = types('session', BASIC_TYPES)


export {
  visitRest,
  encounterRest,
  LOGIN_TYPES,
  SESSION_TYPES
};
