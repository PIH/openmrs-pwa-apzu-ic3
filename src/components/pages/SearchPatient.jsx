import React from 'react';
import { PatientSearch } from '@openmrs/react-components';

class SearchPatient extends React.Component {

    constructor(props) {
        super(props);
        this.columnDefs =  [
            {headerName: 'patientUuid', hide: true, valueGetter: 'data.uuid' },
            {headerName: 'ID', valueGetter: 'data.identifier' },
            {headerName: 'First Name', valueGetter: 'data.firstName' },
            {headerName: 'Last Name', valueGetter: 'data.lastName' },
            {headerName: 'Gender', field: 'gender' },
            {headerName: 'Age', field: 'age' }
        ];
    }


    parseResults(results) {
        console.log("parseResults callback function, length =  " + results.length);
        let patients = [];
        for (const item of results) {
            let patient = {
                uuid: item.uuid,
                id: item.id,
                firstName: item.person.names[0].givenName,
                lastName: item.person.names[0].familyName,
                gender: item.person.gender,
                age: item.person.age,
                identifier: item.identifiers[0].identifier,
                checkedInTime: null,
                birthdate: item.person.birthdate
            };
            patients.push(patient);
        }
        return patients;
    };

    render() {
        return(
            <PatientSearch columnDefs={ this.columnDefs } parseResults={ this.parseResults.bind(this) }/>
        );
    }
};

export default SearchPatient;
