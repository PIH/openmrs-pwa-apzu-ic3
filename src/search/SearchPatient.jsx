import React from 'react';
import { PatientSearch } from '@openmrs/react-components';
import { push } from 'connected-react-router';
import patientActions from '../patient/patientActions';
import IdentifierFilters from '../screening/IdentifierFilters';
import utils from '../utils'


let SearchPatient = (props) => {
  return (
    <div>
      <PatientSearch
        AdditionalFilters={IdentifierFilters}
        rowSelectedActionCreators={[
          patientActions.getIC3PatientScreeningData,
          () => push('/checkin/checkInPage')
        ]}
        getPatientIdentifiers={utils.getPatientIdentifiers}
        title="Search for Patient"
      />
    </div>
  );
};

export default SearchPatient;
