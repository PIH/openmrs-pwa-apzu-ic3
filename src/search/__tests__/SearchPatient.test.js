import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import {DataGrid, PatientSearch, patientActions} from '@openmrs/react-components';
import { spy } from 'sinon';
import SearchPatient from '../SearchPatient';

let props, store;
let mountedComponent;

const mockStore = configureMockStore();

const searchPatient = () => {
  if (!mountedComponent) {
    mountedComponent = mount(
      <Provider store={store}>
        <SearchPatient {...props}></SearchPatient>
      </Provider>);
  }
  return mountedComponent;
};

describe('Component: SearchPatient', () => {
  beforeEach(() => {
    props = {
      columnDefs: []
    };
    store = mockStore(
      {
        dispatch: {},
        openmrs: {
          patientSearch: {
            results: [
              {
                address: "",
                age: 18,
                gender: "M",
                identifiers: [
                  {
                    identifier: "NOP 55 CCC",
                    preferred: true
                  }
                ]
              }
            ]
          },
          selectedPatient: null
        },
      });
    mountedComponent = undefined;
  });

  it('renders properly', () => {
    expect(toJson(searchPatient())).toMatchSnapshot();
    expect(searchPatient().find(PatientSearch).length).toBe(1);
    expect(searchPatient().find(DataGrid).props().rowSelectedActionCreators.length).toBe(4);
    expect(searchPatient().find(DataGrid).props().rowSelectedActionCreators[0].name).toBe("updatePatientInStore");
    expect(searchPatient().find(DataGrid).props().rowSelectedActionCreators[1].name).toBe("setSelectedPatient");
    expect(searchPatient().find(DataGrid).props().rowSelectedActionCreators[2].name).toBe("getPatientApptData");
    expect(searchPatient().find(DataGrid).props().rowSelectedActionCreators[3].name).toBe("");
    expect(searchPatient().find(DataGrid).props().rowSelectedActionCreators[3]().payload.args[0]).toBe("/checkin/checkInPage");
    expect(store.getActions()).toContainEqual(patientActions.clearSelectedPatient());
  });

});

