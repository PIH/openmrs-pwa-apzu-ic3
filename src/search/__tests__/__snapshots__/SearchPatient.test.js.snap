// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SearchPatient renders properly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(SearchPatient)
    columnDefs={Array []}
  >
    <SearchPatient
      columnDefs={Array []}
      dispatch={[Function]}
    >
      <div>
        <h3>
          <Label
            bsClass="label"
            bsStyle="default"
          >
            <span
              className="label label-default"
            >
              Patient Search
            </span>
          </Label>
        </h3>
        <Connect(PatientSearch)
          columnDefs={
            Array [
              Object {
                "field": "uuid",
                "headerName": "uuid",
                "hide": true,
              },
              Object {
                "headerName": "Id",
                "valueGetter": "data.identifiers[0].identifier",
              },
              Object {
                "field": "name.givenName",
                "headerName": "Given Name",
              },
              Object {
                "field": "name.familyName",
                "headerName": "Family Name",
              },
              Object {
                "field": "gender",
                "headerName": "Gender",
              },
              Object {
                "field": "age",
                "headerName": "Age",
              },
            ]
          }
          parseResults={[Function]}
          rowSelectedActionCreators={
            Array [
              [Function],
              [Function],
            ]
          }
        >
          <PatientSearch
            columnDefs={
              Array [
                Object {
                  "field": "uuid",
                  "headerName": "uuid",
                  "hide": true,
                },
                Object {
                  "headerName": "Id",
                  "valueGetter": "data.identifiers[0].identifier",
                },
                Object {
                  "field": "name.givenName",
                  "headerName": "Given Name",
                },
                Object {
                  "field": "name.familyName",
                  "headerName": "Family Name",
                },
                Object {
                  "field": "gender",
                  "headerName": "Gender",
                },
                Object {
                  "field": "age",
                  "headerName": "Age",
                },
              ]
            }
            dispatch={[Function]}
            parseResults={[Function]}
            representation="full"
            rowData={Array []}
            rowSelectedActionCreators={
              Array [
                [Function],
                [Function],
              ]
            }
          >
            <div>
              <ReduxForm
                onSubmit={[Function]}
              >
                <Connect(Form(PatientSearchForm))
                  destroyOnUnmount={true}
                  enableReinitialize={false}
                  forceUnregisterOnUnmount={false}
                  form="search-form"
                  getFormState={[Function]}
                  keepDirtyOnReinitialize={false}
                  onSubmit={[Function]}
                  persistentSubmitErrors={false}
                  pure={true}
                  shouldAsyncValidate={[Function]}
                  shouldError={[Function]}
                  shouldValidate={[Function]}
                  shouldWarn={[Function]}
                  touchOnBlur={true}
                  touchOnChange={false}
                  updateUnregisteredFields={false}
                >
                  <Form(PatientSearchForm)
                    anyTouched={false}
                    array={
                      Object {
                        "insert": [Function],
                        "move": [Function],
                        "pop": [Function],
                        "push": [Function],
                        "remove": [Function],
                        "removeAll": [Function],
                        "shift": [Function],
                        "splice": [Function],
                        "swap": [Function],
                        "unshift": [Function],
                      }
                    }
                    arrayInsert={[Function]}
                    arrayMove={[Function]}
                    arrayPop={[Function]}
                    arrayPush={[Function]}
                    arrayRemove={[Function]}
                    arrayRemoveAll={[Function]}
                    arrayShift={[Function]}
                    arraySplice={[Function]}
                    arraySwap={[Function]}
                    arrayUnshift={[Function]}
                    asyncValidating={false}
                    autofill={[Function]}
                    blur={[Function]}
                    change={[Function]}
                    clearAsyncError={[Function]}
                    clearFields={[Function]}
                    clearSubmit={[Function]}
                    clearSubmitErrors={[Function]}
                    destroy={[Function]}
                    destroyOnUnmount={true}
                    dirty={false}
                    dispatch={[Function]}
                    enableReinitialize={false}
                    focus={[Function]}
                    forceUnregisterOnUnmount={false}
                    form="search-form"
                    getFormState={[Function]}
                    initialize={[Function]}
                    initialized={false}
                    invalid={false}
                    keepDirtyOnReinitialize={false}
                    onSubmit={[Function]}
                    persistentSubmitErrors={false}
                    pristine={true}
                    pure={true}
                    registerField={[Function]}
                    reset={[Function]}
                    resetSection={[Function]}
                    setSubmitFailed={[Function]}
                    setSubmitSucceeded={[Function]}
                    shouldAsyncValidate={[Function]}
                    shouldError={[Function]}
                    shouldValidate={[Function]}
                    shouldWarn={[Function]}
                    startAsyncValidation={[Function]}
                    startSubmit={[Function]}
                    stopAsyncValidation={[Function]}
                    stopSubmit={[Function]}
                    submit={[Function]}
                    submitFailed={false}
                    submitSucceeded={false}
                    submitting={false}
                    syncErrors={Object {}}
                    syncWarnings={Object {}}
                    touch={[Function]}
                    touchOnBlur={true}
                    touchOnChange={false}
                    unregisterField={[Function]}
                    untouch={[Function]}
                    updateSyncErrors={[Function]}
                    updateSyncWarnings={[Function]}
                    updateUnregisteredFields={false}
                    valid={true}
                    validExceptSubmit={true}
                    values={Object {}}
                  >
                    <PatientSearchForm
                      anyTouched={false}
                      array={
                        Object {
                          "insert": [Function],
                          "move": [Function],
                          "pop": [Function],
                          "push": [Function],
                          "remove": [Function],
                          "removeAll": [Function],
                          "shift": [Function],
                          "splice": [Function],
                          "swap": [Function],
                          "unshift": [Function],
                        }
                      }
                      asyncValidate={[Function]}
                      asyncValidating={false}
                      autofill={[Function]}
                      blur={[Function]}
                      change={[Function]}
                      clearAsyncError={[Function]}
                      clearFields={[Function]}
                      clearSubmit={[Function]}
                      clearSubmitErrors={[Function]}
                      destroy={[Function]}
                      dirty={false}
                      dispatch={[Function]}
                      form="search-form"
                      handleSubmit={[Function]}
                      initialize={[Function]}
                      initialized={false}
                      invalid={false}
                      onSubmit={[Function]}
                      pristine={true}
                      pure={true}
                      reset={[Function]}
                      resetSection={[Function]}
                      submit={[Function]}
                      submitFailed={false}
                      submitSucceeded={false}
                      submitting={false}
                      touch={[Function]}
                      untouch={[Function]}
                      valid={true}
                    >
                      <Form
                        bsClass="form"
                        componentClass="form"
                        horizontal={false}
                        inline={false}
                        onSubmit={[Function]}
                      >
                        <form
                          className=""
                          onSubmit={[Function]}
                        >
                          <Grid
                            bsClass="container"
                            componentClass="div"
                            fluid={false}
                          >
                            <div
                              className="container"
                            >
                              <Row
                                bsClass="row"
                                componentClass="div"
                              >
                                <div
                                  className="row"
                                >
                                  <FormGroup
                                    bsClass="form-group"
                                  >
                                    <div
                                      className="form-group"
                                    >
                                      <Col
                                        bsClass="col"
                                        componentClass={[Function]}
                                      >
                                        <ControlLabel
                                          bsClass="control-label"
                                          className=""
                                          srOnly={false}
                                        >
                                          <label
                                            className="control-label"
                                          >
                                            Search
                                          </label>
                                        </ControlLabel>
                                      </Col>
                                      <Col
                                        bsClass="col"
                                        componentClass="div"
                                      >
                                        <div
                                          className=""
                                        >
                                          <Field
                                            component="input"
                                            name="query"
                                            type="text"
                                          >
                                            <Connect(ConnectedField)
                                              _reduxForm={
                                                Object {
                                                  "anyTouched": false,
                                                  "array": Object {
                                                    "insert": [Function],
                                                    "move": [Function],
                                                    "pop": [Function],
                                                    "push": [Function],
                                                    "remove": [Function],
                                                    "removeAll": [Function],
                                                    "shift": [Function],
                                                    "splice": [Function],
                                                    "swap": [Function],
                                                    "unshift": [Function],
                                                  },
                                                  "arrayInsert": [Function],
                                                  "arrayMove": [Function],
                                                  "arrayPop": [Function],
                                                  "arrayPush": [Function],
                                                  "arrayRemove": [Function],
                                                  "arrayRemoveAll": [Function],
                                                  "arrayShift": [Function],
                                                  "arraySplice": [Function],
                                                  "arraySwap": [Function],
                                                  "arrayUnshift": [Function],
                                                  "asyncErrors": undefined,
                                                  "asyncValidate": [Function],
                                                  "asyncValidating": false,
                                                  "autofill": [Function],
                                                  "blur": [Function],
                                                  "change": [Function],
                                                  "clearAsyncError": [Function],
                                                  "clearFields": [Function],
                                                  "clearSubmit": [Function],
                                                  "clearSubmitErrors": [Function],
                                                  "destroy": [Function],
                                                  "destroyOnUnmount": true,
                                                  "dirty": false,
                                                  "dispatch": [Function],
                                                  "enableReinitialize": false,
                                                  "error": undefined,
                                                  "focus": [Function],
                                                  "forceUnregisterOnUnmount": false,
                                                  "form": "search-form",
                                                  "getFormState": [Function],
                                                  "getValues": [Function],
                                                  "initialValues": undefined,
                                                  "initialize": [Function],
                                                  "initialized": false,
                                                  "invalid": false,
                                                  "keepDirtyOnReinitialize": false,
                                                  "onSubmit": [Function],
                                                  "persistentSubmitErrors": false,
                                                  "pristine": true,
                                                  "pure": true,
                                                  "register": [Function],
                                                  "registerField": [Function],
                                                  "registerInnerOnSubmit": [Function],
                                                  "registeredFields": undefined,
                                                  "reset": [Function],
                                                  "resetSection": [Function],
                                                  "sectionPrefix": undefined,
                                                  "setSubmitFailed": [Function],
                                                  "setSubmitSucceeded": [Function],
                                                  "shouldAsyncValidate": [Function],
                                                  "shouldError": [Function],
                                                  "shouldValidate": [Function],
                                                  "shouldWarn": [Function],
                                                  "startAsyncValidation": [Function],
                                                  "startSubmit": [Function],
                                                  "stopAsyncValidation": [Function],
                                                  "stopSubmit": [Function],
                                                  "submit": [Function],
                                                  "submitFailed": false,
                                                  "submitSucceeded": false,
                                                  "submitting": false,
                                                  "syncErrors": Object {},
                                                  "syncWarnings": Object {},
                                                  "touch": [Function],
                                                  "touchOnBlur": true,
                                                  "touchOnChange": false,
                                                  "triggerSubmit": undefined,
                                                  "unregister": [Function],
                                                  "unregisterField": [Function],
                                                  "untouch": [Function],
                                                  "updateSyncErrors": [Function],
                                                  "updateSyncWarnings": [Function],
                                                  "updateUnregisteredFields": false,
                                                  "valid": true,
                                                  "validExceptSubmit": true,
                                                  "values": Object {},
                                                  "warning": undefined,
                                                }
                                              }
                                              component="input"
                                              name="query"
                                              normalize={[Function]}
                                              type="text"
                                            >
                                              <ConnectedField
                                                _reduxForm={
                                                  Object {
                                                    "anyTouched": false,
                                                    "array": Object {
                                                      "insert": [Function],
                                                      "move": [Function],
                                                      "pop": [Function],
                                                      "push": [Function],
                                                      "remove": [Function],
                                                      "removeAll": [Function],
                                                      "shift": [Function],
                                                      "splice": [Function],
                                                      "swap": [Function],
                                                      "unshift": [Function],
                                                    },
                                                    "arrayInsert": [Function],
                                                    "arrayMove": [Function],
                                                    "arrayPop": [Function],
                                                    "arrayPush": [Function],
                                                    "arrayRemove": [Function],
                                                    "arrayRemoveAll": [Function],
                                                    "arrayShift": [Function],
                                                    "arraySplice": [Function],
                                                    "arraySwap": [Function],
                                                    "arrayUnshift": [Function],
                                                    "asyncErrors": undefined,
                                                    "asyncValidate": [Function],
                                                    "asyncValidating": false,
                                                    "autofill": [Function],
                                                    "blur": [Function],
                                                    "change": [Function],
                                                    "clearAsyncError": [Function],
                                                    "clearFields": [Function],
                                                    "clearSubmit": [Function],
                                                    "clearSubmitErrors": [Function],
                                                    "destroy": [Function],
                                                    "destroyOnUnmount": true,
                                                    "dirty": false,
                                                    "dispatch": [Function],
                                                    "enableReinitialize": false,
                                                    "error": undefined,
                                                    "focus": [Function],
                                                    "forceUnregisterOnUnmount": false,
                                                    "form": "search-form",
                                                    "getFormState": [Function],
                                                    "getValues": [Function],
                                                    "initialValues": undefined,
                                                    "initialize": [Function],
                                                    "initialized": false,
                                                    "invalid": false,
                                                    "keepDirtyOnReinitialize": false,
                                                    "onSubmit": [Function],
                                                    "persistentSubmitErrors": false,
                                                    "pristine": true,
                                                    "pure": true,
                                                    "register": [Function],
                                                    "registerField": [Function],
                                                    "registerInnerOnSubmit": [Function],
                                                    "registeredFields": undefined,
                                                    "reset": [Function],
                                                    "resetSection": [Function],
                                                    "sectionPrefix": undefined,
                                                    "setSubmitFailed": [Function],
                                                    "setSubmitSucceeded": [Function],
                                                    "shouldAsyncValidate": [Function],
                                                    "shouldError": [Function],
                                                    "shouldValidate": [Function],
                                                    "shouldWarn": [Function],
                                                    "startAsyncValidation": [Function],
                                                    "startSubmit": [Function],
                                                    "stopAsyncValidation": [Function],
                                                    "stopSubmit": [Function],
                                                    "submit": [Function],
                                                    "submitFailed": false,
                                                    "submitSucceeded": false,
                                                    "submitting": false,
                                                    "syncErrors": Object {},
                                                    "syncWarnings": Object {},
                                                    "touch": [Function],
                                                    "touchOnBlur": true,
                                                    "touchOnChange": false,
                                                    "triggerSubmit": undefined,
                                                    "unregister": [Function],
                                                    "unregisterField": [Function],
                                                    "untouch": [Function],
                                                    "updateSyncErrors": [Function],
                                                    "updateSyncWarnings": [Function],
                                                    "updateUnregisteredFields": false,
                                                    "valid": true,
                                                    "validExceptSubmit": true,
                                                    "values": Object {},
                                                    "warning": undefined,
                                                  }
                                                }
                                                asyncValidating={false}
                                                component="input"
                                                dirty={false}
                                                dispatch={[Function]}
                                                name="query"
                                                normalize={[Function]}
                                                pristine={true}
                                                type="text"
                                              >
                                                <input
                                                  name="query"
                                                  onBlur={[Function]}
                                                  onChange={[Function]}
                                                  onDragStart={[Function]}
                                                  onDrop={[Function]}
                                                  onFocus={[Function]}
                                                  type="text"
                                                  value=""
                                                />
                                              </ConnectedField>
                                            </Connect(ConnectedField)>
                                          </Field>
                                        </div>
                                      </Col>
                                    </div>
                                  </FormGroup>
                                </div>
                              </Row>
                              <Row
                                bsClass="row"
                                componentClass="div"
                              >
                                <div
                                  className="row"
                                >
                                  <FormGroup
                                    bsClass="form-group"
                                  >
                                    <div
                                      className="form-group"
                                    >
                                      <Col
                                        bsClass="col"
                                        componentClass="div"
                                      >
                                        <div
                                          className=""
                                        >
                                          <ButtonToolbar
                                            bsClass="btn-toolbar"
                                          >
                                            <div
                                              className="btn-toolbar"
                                              role="toolbar"
                                            >
                                              <Button
                                                active={false}
                                                block={false}
                                                bsClass="btn"
                                                bsSize="large"
                                                bsStyle="success"
                                                disabled={true}
                                                type="submit"
                                              >
                                                <button
                                                  className="btn btn-lg btn-success"
                                                  disabled={true}
                                                  type="submit"
                                                >
                                                  Search
                                                </button>
                                              </Button>
                                              <Button
                                                active={false}
                                                block={false}
                                                bsClass="btn"
                                                bsSize="large"
                                                bsStyle="danger"
                                                disabled={true}
                                                onClick={[Function]}
                                              >
                                                <button
                                                  className="btn btn-lg btn-danger"
                                                  disabled={true}
                                                  onClick={[Function]}
                                                  type="button"
                                                >
                                                  Clear Values
                                                </button>
                                              </Button>
                                            </div>
                                          </ButtonToolbar>
                                        </div>
                                      </Col>
                                    </div>
                                  </FormGroup>
                                </div>
                              </Row>
                            </div>
                          </Grid>
                        </form>
                      </Form>
                    </PatientSearchForm>
                  </Form(PatientSearchForm)>
                </Connect(Form(PatientSearchForm))>
              </ReduxForm>
              <Connect(DataGrid)
                columnDefs={
                  Array [
                    Object {
                      "field": "uuid",
                      "headerName": "uuid",
                      "hide": true,
                    },
                    Object {
                      "headerName": "Id",
                      "valueGetter": "data.identifiers[0].identifier",
                    },
                    Object {
                      "field": "name.givenName",
                      "headerName": "Given Name",
                    },
                    Object {
                      "field": "name.familyName",
                      "headerName": "Family Name",
                    },
                    Object {
                      "field": "gender",
                      "headerName": "Gender",
                    },
                    Object {
                      "field": "age",
                      "headerName": "Age",
                    },
                  ]
                }
                rowData={Array []}
                rowSelectedActionCreators={
                  Array [
                    [Function],
                    [Function],
                  ]
                }
              >
                <DataGrid
                  columnDefs={
                    Array [
                      Object {
                        "field": "uuid",
                        "headerName": "uuid",
                        "hide": true,
                      },
                      Object {
                        "headerName": "Id",
                        "valueGetter": "data.identifiers[0].identifier",
                      },
                      Object {
                        "field": "name.givenName",
                        "headerName": "Given Name",
                      },
                      Object {
                        "field": "name.familyName",
                        "headerName": "Family Name",
                      },
                      Object {
                        "field": "gender",
                        "headerName": "Gender",
                      },
                      Object {
                        "field": "age",
                        "headerName": "Age",
                      },
                    ]
                  }
                  dispatch={[Function]}
                  rowData={Array []}
                  rowSelectedActionCreators={
                    Array [
                      [Function],
                      [Function],
                    ]
                  }
                >
                  <div
                    style={
                      Object {
                        "height": "600px",
                      }
                    }
                  >
                    <label>
                      Search:
                      <input
                        onChange={[Function]}
                        type="text"
                      />
                    </label>
                    <AgGridReact
                      columnDefs={
                        Array [
                          Object {
                            "field": "uuid",
                            "headerName": "uuid",
                            "hide": true,
                          },
                          Object {
                            "headerName": "Id",
                            "valueGetter": "data.identifiers[0].identifier",
                          },
                          Object {
                            "field": "name.givenName",
                            "headerName": "Given Name",
                          },
                          Object {
                            "field": "name.familyName",
                            "headerName": "Family Name",
                          },
                          Object {
                            "field": "gender",
                            "headerName": "Gender",
                          },
                          Object {
                            "field": "age",
                            "headerName": "Age",
                          },
                        ]
                      }
                      enableSorting={true}
                      id="omrsGrid"
                      onGridReady={[Function]}
                      onSelectionChanged={[Function]}
                      rowClassRules="rowClassRules"
                      rowData={Array []}
                      rowSelection="single"
                    >
                      <div
                        style={
                          Object {
                            "height": "100%",
                          }
                        }
                      />
                    </AgGridReact>
                  </div>
                </DataGrid>
              </Connect(DataGrid)>
            </div>
          </PatientSearch>
        </Connect(PatientSearch)>
      </div>
    </SearchPatient>
  </Connect(SearchPatient)>
</Provider>
`;
